File: c/puzzles/fizzbuzz.c
#include <stdio.h>

void main(void) {
    unsigned int short i, fizz, buzz;

    for (i = 0x1; i <= 0x64; i++) {
        fizz = buzz = 0x0;

        if (i % 0x3 == 0x0) {
            fizz = 0x3;
            printf("fizz, number %d\n", i);
        }

        if (i % 0x5 == 0x0) {
            buzz = 0x5;
            printf("buzz, number %d\n", i);
        }

        // if (i % 0xf == 0x0) {
        if ((fizz & buzz) == 0x1) {
            printf("fizzbuzz, number %d\n", i);
        }
    }
}

File: javascript/puzzles/fizzbuzz.js
for (let i = 1; i <= 20; i++) {
    let fizzbuzz = 0;
//
//    s = ""
//    if (i % 3 == 0) {
//        s += "fizz"
//    }
//
//    if (i % 5 == 0) {
//        s += "buzz"
//    }
//
//    console.log(i, s)

    if (i % 3 === 0) {
        fizzbuzz = 1;
    }

    if (i % 5 === 0) {
        fizzbuzz += 2;
    }

    switch (fizzbuzz & 3) {
        case 0:
            console.log(i);
            break;
        case 1:
            console.log('fizz');
            break;
        case 2:
            console.log('buzz');
            break;
        case 3:
            console.log('fizzbuzz');
            break;
    }
}

